tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Inheritance.java

```diff
@@ -122,38 +124,45 @@ public Response getInheritanceByHex(
         return Response.ok(new InheritanceResponse(paths)).build();
     }
 
-    private String getXPathByHex(String hexId) {
-        String xpath = null;
-        try {
-            xpath = sm.xpt.getByStringID(hexId);
-        } catch (RuntimeException e) {
-            /*
-             * Don't report the exception. This happens when it simply wasn't found.
-             * Possibly getByStringID, or some version of it, should not throw an exception.
-             */
-        }
-        return xpath;
-    }
-
     /** Streamable version of {@link LocaleInheritanceInfo} */
     public static final class LocaleInheritance {
+        @Schema(description = "xpath stringid to item")
         public String xpath;
+
+        @Schema(description = "locale of item if present")
         public String locale;
+
+        @Schema(description = "xpath full of item")
         public String xpathFull;
+
+        @Schema(description = "which attribute was implicated in a change")
         public String attribute;
+
+        @Schema(description = "reason for the entry")
         public LocaleInheritanceInfo.Reason reason;
 
-        public LocaleInheritance(LocaleInheritanceInfo info) {
+        @Schema(description = "true if not shown in the SurveyTool")
+        public boolean hidden = false;
+
+        public LocaleInheritance(LocaleInheritanceInfo info, final STFactory stf) {
             this.reason = info.getReason();
             this.locale = info.getLocale();
             this.attribute = info.getAttribute();
             final String x = info.getPath();
             if (x != null) {
                 this.xpath = StringId.getHexId(x);
+                CookieSession.sm.xpt.getByXpath(x); // make sure it's in the XPT!
```

Comment:
this is a little mysterious to me; supposing bogus http requests contained bogus xpaths, would they get added to the db?